source/CScriptEngine.cpp

```diff
@@ -737,10 +740,10 @@ namespace CLEO
     {
         std::ostringstream ss;
 
-        auto threadName = GetName();
+        auto threadName = std::string(GetName(), GetName() + 8); // thread name might not be null terminated

```

Comment:
what about other places where we expect a null-terminated string?

e.g. https://github.com/cleolibrary/CLEO5/blob/master/source/CCustomOpcodeSystem.cpp#L1432

Reply:
Right. Perhaps GetName() function should always return c-string.